components/Buttons/src/Theming/MDCFloatingButton+Theming.m

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCFloatingButton+Theming.h"
+
+#import "MaterialButtons+ColorThemer.h"
+#import "MaterialButtons+ShapeThemer.h"
+#import "MaterialButtons+TypographyThemer.h"
+
+@implementation MDCFloatingButton (MaterialTheming)
+
+- (void)applySecondaryThemeWithScheme:(id<MDCContainerScheming>)scheme {
+  id<MDCColorScheming> colorScheme = scheme.colorScheme;
+  if (!colorScheme) {
+    colorScheme =
+        [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  }
+  [self applySecondaryThemeWithColorScheme:colorScheme];
+
+  id<MDCShapeScheming> shapeScheme = scheme.shapeScheme;
+  if (!shapeScheme) {
+    shapeScheme = [[MDCShapeScheme alloc] initWithDefaults:MDCShapeSchemeDefaultsMaterial201809];
+  }
+  [self applySecondaryThemeWithShapeScheme:shapeScheme];
+
+  id<MDCTypographyScheming> typographyScheme = scheme.typographyScheme;
+  if (!typographyScheme) {
+    typographyScheme =
+        [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  }
+  [self applySecondaryThemeWithTypographyScheme:typographyScheme];
+
+  [self setElevation:(CGFloat)6 forState:UIControlStateNormal];
```

Comment:
Can MDCShadowElevations be used instead of literals?

Reply:
This is no corresponding MDCShadowElevation defined for this value, and from this PR's purpose, it is okay if we don't do the constant refactoring right now.

Reply:
Please use `MDCShadowElevationFABResting` as it is the correct elevation for the `resting` (`.normal`) state of the Floating Action Button (FAB).

Reply:
Done.